analysis_driver/dataset.py

```diff
@@ -293,7 +293,8 @@ def run_info(self):
     def sample_sheet_file(self):
         if self._sample_sheet_file is None:
             self._sample_sheet_file = os.path.join(self.input_dir, 'SampleSheet_analysis_driver.csv')
-            self._generate_samplesheet(self._sample_sheet_file)
+            if not os.path.isfile(self._sample_sheet_file):
```

Comment:
@tcezard So the file is no longer generated if it already exists? Could this be a problem if we need to repeat the sample sheet file generation and it already exists?

Reply:
There might be cases in the future when we will need to remove an erroneous samplesheet to make sure it gets regenerated.

Reply:
May I ask why we are regenerating every time this function is called, as was happening previously?

Reply:
We could now that the generation is deterministic but there is no need. The Samplesheet is not suppose to change